File: GZCTF/Models/Request/Edit/ChallengeInfoModel.cs
using System.ComponentModel.DataAnnotations;

namespace CTFServer.Models.Request.Edit;

/// <summary>
/// 基础题目信息（Edit）
/// </summary>
public class ChallengeInfoModel
{
    /// <summary>
    /// 题目Id
    /// </summary>
    public int Id { get; set; }

    /// <summary>
    /// 题目名称
    /// </summary>
    [Required(ErrorMessage = "标题是必需的")]
    [MinLength(1, ErrorMessage = "标题过短")]
    public string Title { get; set; } = string.Empty;

    /// <summary>
    /// 题目标签
    /// </summary>
    public ChallengeTag Tag { get; set; } = ChallengeTag.Misc;

    /// <summary>
    /// 题目类型
    /// </summary>
    public ChallengeType Type { get; set; } = ChallengeType.StaticAttachment;

    /// <summary>
    /// 是否启用题目
    /// </summary>
    public bool IsEnabled { get; set; } = false;

    public static ChallengeInfoModel FromChallenge(Challenge challenge)
        => new()
        {
            Id = challenge.Id,
            Title = challenge.Title,
            Tag = challenge.Tag,
            Type = challenge.Type,
            IsEnabled = challenge.IsEnabled
        };
}

File: GZCTF/ClientApp/pages/client/models/ChallengeInfoModel.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { ChallengeTag } from './ChallengeTag';
import type { ChallengeType } from './ChallengeType';

export type ChallengeInfoModel = {
    /**
     * 题目Id
     */
    id?: number;
    /**
     * 题目名称
     */
    title: string;
    /**
     * 题目标签
     */
    tag?: ChallengeTag;
    /**
     * 题目类型
     */
    type?: ChallengeType;
};
